habitat-lab/habitat/tasks/rearrange/rearrange_grasp_manager.py

```diff
@@ -323,6 +345,14 @@ def snap_to_obj(
         self._managed_articulated_agent.open_gripper()
 
         if self._kinematic_mode:
+            # update the KRM to sever any existing parent relationships for the newly grasped object
+            self._sim.kinematic_relationship_manager.relationship_graph.remove_obj_relations(
+                snap_obj_id, parents_only=True
+            )
+            # update root parent transforms so new parent state is registered
```

Comment:
Why do we need this? I understand that the line above removes the relationship with the parent. Won't the root parent transforms stay the same when we pick an object? Why do we need to store them in prev_root_obj_state?

Reply:
When we disconnect an object from its parents, the object becomes a root parent. We need to cache the new root parent's global transform so it can be applied to update any children. 

This is the essential logic which allows, for example, picking up a tray with cups on it and moving the cups along with the tray. The tray is a new root parent and its transform must be tracked.